PyPDF2/_writer.py

```diff
@@ -868,8 +869,13 @@ def _sweep_indirect_references(
                 if newobj is None:
                     try:
                         newobj = data.pdf.get_object(data)
+                        hash_value = newobj.hash_value()
+                        # Check if object is already added to pdf.
+                        if hash_value in self._idnum_hash:
+                            return IndirectObject(self._idnum_hash[hash_value], 0, self)
                         self._objects.append(None)  # placeholder
                         idnum = len(self._objects)
+                        self._idnum_hash[hash_value] = idnum
```

Comment:
Should we be doing a similar operation [here](https://github.com/py-pdf/PyPDF2/blob/fe7b4c9df5bd65be7eeeee252ebea4862389f3e9/PyPDF2/_writer.py#L734) where the `extern_map` object is original setup?

Reply:
I'm not sure.

Reply:
@MasterOdin I think this PR already adds quite a bit of value and Hatell already rebased twice due to other changes. In order to Hatells side low, I tend to merge soon without the de-duplicatoin with extern_map. We can still do it later (depending on how complex it is, I might do it directly after merging). Would that be ok for you?

Reply:
That's fine.